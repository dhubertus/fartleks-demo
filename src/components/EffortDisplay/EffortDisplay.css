#effort-card-display {
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: 140px;
  border-top: 1px solid #3d3e42;
  border-bottom: 1px solid #3d3e42;
  background: #edeef0;
  flex-direction: column;

}

#effort-display-title {
  height:22%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #3d3e42;
  color: #00aced;
  /* box-shadow: 1px 1px 1px 1px black; */
}

#effort-display-title h4 {
  margin: 0;
  font-weight: 100;
}

#effort-card-container {
  display: flex;
  justify-content: space-around;
  width: 100%;
  height: 78%;
  align-items: center;
}
