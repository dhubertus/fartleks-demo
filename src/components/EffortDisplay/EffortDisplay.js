import React, { Component } from 'react';

import EffortCard from '../EffortCard/EffortCard';
import BestEffortsContainer from '../../containers/BestEffortsContainer';

export class EffortDisplay extends Component {
  render() {
    const { bestEfforts } = this.props;
    const cardArray = Object.keys(bestEfforts).map((key, i) => {
      return (
        <EffortCard
          key={i}
          title={bestEfforts[key].title}
          bestAverage={bestEfforts[key].bestAverage}
          bestValues={bestEfforts[key].bestValues}
        />
      );
    });

    return (
      <div id="effort-card-display">
        <div id='effort-display-title'>
          <h4>Your Best Efforts</h4>
        </div>
        <div id='effort-card-container'>
          {cardArray}
        </div>
      </div>
    );
  }
}

export default BestEffortsContainer(EffortDisplay);
