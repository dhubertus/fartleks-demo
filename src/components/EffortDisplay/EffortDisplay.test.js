import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { EffortDisplay } from './EffortDisplay';

describe('Effort Display Tests', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = {
    selectedData: [],
    bestEfforts: {},
  };
  const bestEfforts = {
    one: {},
    five: {},
    ten: {},
    fifteen: {},
    twenty: {},
  };
  const store = mockStore(initialState);
  const wrapper = shallow(<EffortDisplay
                            store={store}
                            bestEfforts={bestEfforts}
                          />);

  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a single display', () => {
    expect(wrapper.find('#effort-card-display').length).toBe(1);
  });

  it('should render a single display title', () => {
    expect(wrapper.find('#effort-display-title').length).toBe(1);
  });

  it('should render 5 effort cards when passed 5 data points', () => {
    expect(wrapper.find('#effort-card-container').children().length).toBe(5);
  });
});
