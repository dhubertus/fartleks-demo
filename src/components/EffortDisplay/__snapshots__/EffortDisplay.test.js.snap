// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Effort Display Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EffortDisplay
    bestEfforts={
        Object {
            "fifteen": Object {},
            "five": Object {},
            "one": Object {},
            "ten": Object {},
            "twenty": Object {},
          }
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="effort-display-title"
>
          <h4>
                    Your Best Efforts
          </h4>
</div>,
        <div
          id="effort-card-container"
>
          <Connect(EffortCard)
                    bestAverage={undefined}
                    bestValues={undefined}
                    title={undefined}
          />
          <Connect(EffortCard)
                    bestAverage={undefined}
                    bestValues={undefined}
                    title={undefined}
          />
          <Connect(EffortCard)
                    bestAverage={undefined}
                    bestValues={undefined}
                    title={undefined}
          />
          <Connect(EffortCard)
                    bestAverage={undefined}
                    bestValues={undefined}
                    title={undefined}
          />
          <Connect(EffortCard)
                    bestAverage={undefined}
                    bestValues={undefined}
                    title={undefined}
          />
</div>,
      ],
      "id": "effort-card-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            Your Best Efforts
</h4>,
          "id": "effort-display-title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Best Efforts",
          },
          "ref": null,
          "rendered": "Your Best Efforts",
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(EffortCard)
              bestAverage={undefined}
              bestValues={undefined}
              title={undefined}
/>,
            <Connect(EffortCard)
              bestAverage={undefined}
              bestValues={undefined}
              title={undefined}
/>,
            <Connect(EffortCard)
              bestAverage={undefined}
              bestValues={undefined}
              title={undefined}
/>,
            <Connect(EffortCard)
              bestAverage={undefined}
              bestValues={undefined}
              title={undefined}
/>,
            <Connect(EffortCard)
              bestAverage={undefined}
              bestValues={undefined}
              title={undefined}
/>,
          ],
          "id": "effort-card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "bestAverage": undefined,
              "bestValues": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "bestAverage": undefined,
              "bestValues": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "bestAverage": undefined,
              "bestValues": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "bestAverage": undefined,
              "bestValues": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "class",
            "props": Object {
              "bestAverage": undefined,
              "bestValues": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="effort-display-title"
>
            <h4>
                        Your Best Efforts
            </h4>
</div>,
          <div
            id="effort-card-container"
>
            <Connect(EffortCard)
                        bestAverage={undefined}
                        bestValues={undefined}
                        title={undefined}
            />
            <Connect(EffortCard)
                        bestAverage={undefined}
                        bestValues={undefined}
                        title={undefined}
            />
            <Connect(EffortCard)
                        bestAverage={undefined}
                        bestValues={undefined}
                        title={undefined}
            />
            <Connect(EffortCard)
                        bestAverage={undefined}
                        bestValues={undefined}
                        title={undefined}
            />
            <Connect(EffortCard)
                        bestAverage={undefined}
                        bestValues={undefined}
                        title={undefined}
            />
</div>,
        ],
        "id": "effort-card-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              Your Best Efforts
</h4>,
            "id": "effort-display-title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Best Efforts",
            },
            "ref": null,
            "rendered": "Your Best Efforts",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(EffortCard)
                bestAverage={undefined}
                bestValues={undefined}
                title={undefined}
/>,
              <Connect(EffortCard)
                bestAverage={undefined}
                bestValues={undefined}
                title={undefined}
/>,
              <Connect(EffortCard)
                bestAverage={undefined}
                bestValues={undefined}
                title={undefined}
/>,
              <Connect(EffortCard)
                bestAverage={undefined}
                bestValues={undefined}
                title={undefined}
/>,
              <Connect(EffortCard)
                bestAverage={undefined}
                bestValues={undefined}
                title={undefined}
/>,
            ],
            "id": "effort-card-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "bestAverage": undefined,
                "bestValues": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "bestAverage": undefined,
                "bestValues": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "bestAverage": undefined,
                "bestValues": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "class",
              "props": Object {
                "bestAverage": undefined,
                "bestValues": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "class",
              "props": Object {
                "bestAverage": undefined,
                "bestValues": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
